### ISSUE TITLE
pop-upgrade fails if you have removed and blocked snapd from being installed.

### ISSUE THREAD

**Issue by NathanaelA** [pop-os/upgrade#259]
If you block snapd usage (because you don't like snap).   

Pop-upgrade will fail with the following error:
```
Release upgrade status: release upgrade aborted: failed to get transitional snap packages
pop-upgrade: calling ReleaseUpgradeFinalize method failed
```

To Block snap create a `nosnap` file in your /etc/apt/preferences.d folder after you de-install it with the following contents:
```
Package: snapd
Pin: release a=*
Pin-Priority: -10
```

This will keep anything that requires snap from being installed.    

Not sure why the requirement for snapd in 21.10 yet, as I'm still installing it.   But I sincerely hope it isn't a requirement in 21.10, I grab my chromium from a different supported repo that doesn't require snapd.

### Workaround:
Comment out each of those three lines with a `#` before starting the upgrade.

**Comment by jacobgkau** [comment]
> Not sure why the requirement for snapd in 21.10 yet

There is no requirement. In fact, I think [the function attempting to hold transitional snap packages](https://github.com/pop-os/upgrade/blob/788fc604e96caf0c130dc7f85535461112fb74d6/src/release/snapd.rs#L9) exists specifically to prevent snapd from being installed as part of the upgrade in cases where Ubuntu tried to transition a native Debian package to snap (e.g. Chromium.)

**Comment by NathanaelA** [comment]
I can concur; my machine is finally done upgrading and no "snapd" was installed for me to have to de-install -- so not sure why the upgrade required me to unblock snapd installation to work...

**Comment by jacobgkau** [comment]
It's because your pin preference kept the `apt-mark hold snapd ...` command from working.

```
jacob@serw12:~$ sudo apt-mark hold snapd
[sudo] password for jacob: 
snapd set on hold.
jacob@serw12:~$ sudo apt-mark unhold snapd
Canceled hold on snapd.
jacob@serw12:~$ sudo nano /etc/apt/preferences.d/snaptmp
jacob@serw12:~$ sudo apt-mark hold snapd
E: Can't select installed nor candidate version from package 'snapd' as it has neither of them
E: No packages found
```

Some potential ways to handle this:
- Resetting apt preferences to default on release upgrade similar to how apt sources are reset to default
- Ignoring this error from `apt-mark hold` (all the time or only for the snapd operation)

The former seems cleaner to me. We already [apply the default preferences](https://github.com/pop-os/upgrade/blob/788fc604e96caf0c130dc7f85535461112fb74d6/src/release/repos.rs#L281), but we do not currently remove non-default preferences, even though they could conflict with or override the default preferences that we rely on. But I could imagine arguments for the latter as well.

**Comment by NathanaelA** [comment]
> It's because your pin preference kept the apt-mark hold snapd ... command from working.

Oh, I understand the underlying reason.    Just If it isn't needed then, this preference shouldn't actually break things.   ;-)         

As for the solutions, it probably would be the best idea to rename them with .save like the pacakges lists  -- as I'm sure other "preferences" can also break things.  ;-)
